integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonFilters.scala

```diff
@@ -395,4 +391,14 @@ object CarbonFilters {
       case _ => expressions
     }
   }
+
+  def getPartitions(partitionFilters: Seq[Expression],
+      sparkSession: SparkSession,
+      identifier: TableIdentifier): Seq[String] = {
+    val partitions =
+      sparkSession.sessionState.catalog.listPartitionsByFilter(identifier, partitionFilters)
+    partitions.toList.flatMap { f =>
```

Comment:
can you use case(xxx, xxx) to replace `f`, make it more readable

Reply:
ok